pkg/core/region.go

```diff
@@ -1464,6 +1464,97 @@ func (r *RegionsInfo) GetStoreRegions(storeID uint64) []*RegionInfo {
 	return regions
 }
 
+// TODO: benchmark the performance of `QueryRegions`.
+// QueryRegions searches RegionInfo from regionTree by keys and IDs in batch.
+func (r *RegionsInfo) QueryRegions(
```

Comment:
Will we pass both keys and prevKeys at the same time?

Reply:
Yes, if the `GetRegion` and `GetPrevRegion` are batched in the same request.